### ISSUE TITLE
Uncaught TypeError: Cannot read property 'hasOwnProperty' of undefined

### ISSUE THREAD

**Issue by ruicosta042** [atom/atom#9299]
1. I get this bug whenever I start 1.1.0-beta1

**Atom Version**: 1.1.0-beta1
**System**: Mac OS X 10.11.1
**Thrown From**: Atom Core
### Stack Trace

Uncaught TypeError: Cannot read property 'hasOwnProperty' of undefined

```
At /Applications/Atom Beta.app/Contents/Resources/app.asar/src/line-numbers-tile-component.js:88

TypeError: Cannot read property 'hasOwnProperty' of undefined
  at LineNumbersTileComponent.module.exports.LineNumbersTileComponent.updateLineNumbers (/Applications/Atom Beta.app/Contents/Resources/app.asar/src/line-numbers-tile-component.js:88:43)
  at LineNumbersTileComponent.module.exports.LineNumbersTileComponent.updateSync (/Applications/Atom Beta.app/Contents/Resources/app.asar/src/line-numbers-tile-component.js:78:19)
  at LineNumberGutterComponent.module.exports.TiledComponent.updateTileNodes (/Applications/Atom Beta.app/Contents/Resources/app.asar/src/tiled-component.js:67:19)
  at LineNumberGutterComponent.module.exports.TiledComponent.updateSync (/Applications/Atom Beta.app/Contents/Resources/app.asar/src/tiled-component.js:30:12)
  at GutterContainerComponent.module.exports.GutterContainerComponent.updateSync (/Applications/Atom Beta.app/Contents/Resources/app.asar/src/gutter-container-component.js:77:25)
  at TextEditorComponent.module.exports.TextEditorComponent.updateSync (/Applications/Atom Beta.app/Contents/Resources/app.asar/src/text-editor-component.js:221:39)
  at /Applications/Atom Beta.app/Contents/Resources/app.asar/src/text-editor-component.js:298:28
  at ViewRegistry.module.exports.ViewRegistry.performDocumentUpdate (/Applications/Atom Beta.app/Contents/Resources/app.asar/src/view-registry.js:174:9)
  at /Applications/Atom Beta.app/Contents/Resources/app.asar/src/view-registry.js:3:61

```
### Commands

```
```
### Config

``` json
{
  "core": {
    "disabledPackages": [
      "nuclide",
      "nuclide-file-tree",
      "filesize"
    ]
  },
  "editor": {
    "invisibles": {},
    "tabLength": 4
  }
}
```
### Installed Packages

``` coffee
# User
atomdo, v0.8.0
auto-update-packages, v1.0.0
autoclose-html, v0.19.0
autocomplete-paths, v1.0.2
build, v0.48.0
editorconfig, v1.2.0
es-identifier-highlight, v0.2.7
file-icons, v1.6.11
flex-tool-bar, v0.6.0
highlight-line, v0.11.0
highlight-selected, v0.10.1
linter, v1.9.1
linter-jshint, v1.2.1
pigments, v0.18.0
project-manager, v2.6.5
regex-railroad-diagram, v0.10.4
selection-highlight, v0.1.4
terminal-plus, v0.12.2
tool-bar, v0.1.9
tree-view-git-status, v0.2.2

# Dev
No dev packages
```

**Comment by izuzak** [comment]
Thanks for the report, @ruicosta042. Can you reproduce this problem if you start Atom beta in [safe mode](https://atom.io/docs/latest/hacking-atom-debugging#check-if-the-problem-shows-up-in-safe-mode)?

Also cc-ing @as-cii who likes hearing about these kinds of exceptions. :smile:

**Comment by ruicosta042** [comment]
hey, no problem. Can't reproduce it this morning. Maybe this was caused last night by heavy install/uninstall of packages. Sorry for the inconvenience, but it was 100% reproducible for a while, now it's not.

**Comment by izuzak** [comment]
Interesting, thanks for letting us know, @ruicosta042. 

@as-cii I'm going to leave this open until you've had the chance to take a look, but please feel free to close if you think this is a package causing problems.

**Comment by as-cii** [comment]
> who likes hearing about these kinds of exceptions.

:joy: 

> I get this bug whenever I start 1.1.0-beta1

Interesting, I wonder if this could be related to nuclide somehow? Having an environment in which we could reproduce this consistency would be amazing. :sparkles:

**Comment by as-cii** [comment]
Closing this for now as I feel like this is related to a package. @ruicosta042: please, let me know if this happens again and I'll dig a bit more deeply to understand what's causing it.

Thanks! :pray:

**Comment by jdsimcoe** [comment]
:+1: I just got this error when pasting something in a `.jade` file.

**Comment by as-cii** [comment]
Thanks for the report, @jdsimcoe! Can you reproduce it following the steps @izuzak provided a couple of posts above?

**Comment by AndreasHassing** [comment]
Getting the same error, included stacktrace below. First time seeing this, it keeps coming even though I close and reopen Atom.

1. Started Atom after having it closed for 5 minutes
2. Get error

**Atom Version**: 1.1.0
**System**: Mac OS X 10.11.1
**Thrown From**: Atom Core


### Stack Trace

Uncaught TypeError: Cannot read property 'hasOwnProperty' of undefined

```
At /Applications/Atom.app/Contents/Resources/app.asar/src/line-numbers-tile-component.js:88

TypeError: Cannot read property 'hasOwnProperty' of undefined
  at LineNumbersTileComponent.module.exports.LineNumbersTileComponent.updateLineNumbers (/Applications/Atom.app/Contents/Resources/app.asar/src/line-numbers-tile-component.js:88:43)
  at LineNumbersTileComponent.module.exports.LineNumbersTileComponent.updateSync (/Applications/Atom.app/Contents/Resources/app.asar/src/line-numbers-tile-component.js:78:19)
  at LineNumberGutterComponent.module.exports.TiledComponent.updateTileNodes (/Applications/Atom.app/Contents/Resources/app.asar/src/tiled-component.js:67:19)
  at LineNumberGutterComponent.module.exports.TiledComponent.updateSync (/Applications/Atom.app/Contents/Resources/app.asar/src/tiled-component.js:30:12)
  at GutterContainerComponent.module.exports.GutterContainerComponent.updateSync (/Applications/Atom.app/Contents/Resources/app.asar/src/gutter-container-component.js:77:25)
  at TextEditorComponent.module.exports.TextEditorComponent.updateSync (/Applications/Atom.app/Contents/Resources/app.asar/src/text-editor-component.js:221:39)
  at /Applications/Atom.app/Contents/Resources/app.asar/src/text-editor-component.js:298:28
  at ViewRegistry.module.exports.ViewRegistry.performDocumentUpdate (/Applications/Atom.app/Contents/Resources/app.asar/src/view-registry.js:174:9)
  at /Applications/Atom.app/Contents/Resources/app.asar/src/view-registry.js:3:61

```

### Config

```json
{
  "core": {
    "ignoredNames": [
      ".git",
      ".svn",
      ".DS_Store"
    ],
    "disabledPackages": [
      "autosave",
      "open-last-project",
      "travis-ci-status"
    ],
    "themes": [
      "atom-material-ui",
      "atom-material-syntax"
    ],
    "projectHome": "/Users/abn/go",
    "openEmptyEditorOnStart": false
  },
  "editor": {
    "showIndentGuide": true,
    "invisibles": {},
    "softWrapAtPreferredLineLength": true,
    "showInvisibles": true,
    "softWrap": true,
    "fontSize": 15
  }
}
```

### Installed Packages

```coffee
# User
atom-material-syntax, v0.2.4
atom-material-ui, v0.7.1
autocomplete-python, v0.10.0
build, v0.48.0
color-picker, v2.0.13
editorconfig, v1.2.2
go-plus, v3.5.1
language-latex, v0.6.1
linter, v1.10.0
markdown-preview-plus, v2.2.2
mathjax-wrapper, v0.1.0
python-yapf, v0.9.0
vim-mode, v0.63.0
```

**Comment by as-cii** [comment]
@AndreasHassing: can you reproduce in safe mode (i.e. `atom --safe`)? I haven't been able to reproduce this so far, unfortunately.

I am going to keep this closed for the time being, as I'd like to make sure it's not a package causing the problem. Any help in tracking some reproductions steps down would be awesome. :pray: :sparkles:

**Comment by AndreasHassing** [comment]
"Unfortunately" the bug stopped occurring when I closed a particular file that was set to open after launching Atom, and I can't get the bug to come back :/ this was prior to any attempts with safe mode.

**Comment by austinkelleher** [comment]
:+1: I got this error too. Happened when I simply saved a file.

**Comment by MStrohmeier** [comment]
Also getting this issue in Atom on Windows 10. This occurred when I added a new line to the bottom of a file immediately after saving it and Atom removed some extra blank lines at the bottom of the file.

**Comment by DawsonG** [comment]
I started receiving this error every time I scroll to the bottom of one particular file.  The error goes away when I add a blank new line at the end of the file.

I've attached a .txt version of the file, though the original extension was .html.erb
[_form.txt](https://github.com/atom/atom/files/35857/_form.txt)

**Comment by c-kazanci** [comment]
[Enter steps to reproduce below:]

1. ...
2. ...

**Atom Version**: 1.1.0
**System**: Mac OS X 10.11.1
**Thrown From**: Atom Core


### Stack Trace

Uncaught TypeError: Cannot read property 'hasOwnProperty' of undefined

```
At /Applications/Atom.app/Contents/Resources/app.asar/src/line-numbers-tile-component.js:88

TypeError: Cannot read property 'hasOwnProperty' of undefined
    at LineNumbersTileComponent.module.exports.LineNumbersTileComponent.updateLineNumbers (/Applications/Atom.app/Contents/Resources/app.asar/src/line-numbers-tile-component.js:88:43)
    at LineNumbersTileComponent.module.exports.LineNumbersTileComponent.updateSync (/Applications/Atom.app/Contents/Resources/app.asar/src/line-numbers-tile-component.js:78:19)
    at LineNumberGutterComponent.module.exports.TiledComponent.updateTileNodes (/Applications/Atom.app/Contents/Resources/app.asar/src/tiled-component.js:67:19)
    at LineNumberGutterComponent.module.exports.TiledComponent.updateSync (/Applications/Atom.app/Contents/Resources/app.asar/src/tiled-component.js:30:12)
    at GutterContainerComponent.module.exports.GutterContainerComponent.updateSync (/Applications/Atom.app/Contents/Resources/app.asar/src/gutter-container-component.js:77:25)
    at TextEditorComponent.module.exports.TextEditorComponent.updateSync (/Applications/Atom.app/Contents/Resources/app.asar/src/text-editor-component.js:221:39)
    at /Applications/Atom.app/Contents/Resources/app.asar/src/text-editor-component.js:298:28
    at ViewRegistry.module.exports.ViewRegistry.performDocumentUpdate (/Applications/Atom.app/Contents/Resources/app.asar/src/view-registry.js:174:9)
    at /Applications/Atom.app/Contents/Resources/app.asar/src/view-registry.js:3:61
```

### Commands

```
     -0:11 editor:newline (atom-text-editor.editor.is-focused)
  3x -0:08.2.0 core:backspace (atom-text-editor.editor.is-focused)
     -0:02.8.0 autocomplete-plus:confirm (atom-text-editor.editor.is-focused.autocomplete-active)
     -0:00.8.0 editor:newline (atom-text-editor.editor.is-focused)
```

### Config

```json
{
  "core": {
    "themes": [
      "atom-dark-ui",
      "monokai"
    ]
  },
  "editor": {
    "invisibles": {},
    "showIndentGuide": true,
    "zoomFontWhenCtrlScrolling": true,
    "fontSize": 20
  }
}
```

### Installed Packages

```coffee
# User
atom-autocomplete-php, v0.18.1
autocomplete-php, v0.3.7
minimap, v4.16.0
monokai, v0.18.0

# Dev
No dev packages
```

**Comment by nycdotnet** [comment]
For what it's worth, I just got this on Windows with Atom 1.2.4.  I am working in a Markdown file with Soft Wrap on and working toward the bottom-right of the screen and the end of the file.  It felt related to Atom perhaps wrapping to the new virtual line at the end of the file.  I'll report more details if I can get it to happen again.

**Comment by nycdotnet** [comment]
It definitely happened as I was typing across a softwrap at the end of a line in a Markdown file.

[Enter steps to reproduce below:]

1. ...
2. ...

**Atom Version**: 1.2.4
**System**: Microsoft Windows 8.1 Pro
**Thrown From**: Atom Core


### Stack Trace

Uncaught TypeError: Cannot read property 'hasOwnProperty' of undefined

```
At C:\Users\steveo\AppData\Local\atom\app-1.2.4\resources\app.asar\src\line-numbers-tile-component.js:88

TypeError: Cannot read property 'hasOwnProperty' of undefined
    at LineNumbersTileComponent.module.exports.LineNumbersTileComponent.updateLineNumbers (C:\Users\steveo\AppData\Local\atom\app-1.2.4\resources\app.asar\src\line-numbers-tile-component.js:88:43)
    at LineNumbersTileComponent.module.exports.LineNumbersTileComponent.updateSync (C:\Users\steveo\AppData\Local\atom\app-1.2.4\resources\app.asar\src\line-numbers-tile-component.js:78:19)
    at LineNumberGutterComponent.module.exports.TiledComponent.updateTileNodes (C:\Users\steveo\AppData\Local\atom\app-1.2.4\resources\app.asar\src\tiled-component.js:67:19)
    at LineNumberGutterComponent.module.exports.TiledComponent.updateSync (C:\Users\steveo\AppData\Local\atom\app-1.2.4\resources\app.asar\src\tiled-component.js:30:12)
    at GutterContainerComponent.module.exports.GutterContainerComponent.updateSync (C:\Users\steveo\AppData\Local\atom\app-1.2.4\resources\app.asar\src\gutter-container-component.js:79:25)
    at TextEditorComponent.module.exports.TextEditorComponent.updateSync (C:\Users\steveo\AppData\Local\atom\app-1.2.4\resources\app.asar\src\text-editor-component.js:224:39)
    at C:\Users\steveo\AppData\Local\atom\app-1.2.4\resources\app.asar\src\text-editor-component.js:302:28
    at ViewRegistry.module.exports.ViewRegistry.performDocumentUpdate (C:\Users\steveo\AppData\Local\atom\app-1.2.4\resources\app.asar\src\view-registry.js:177:9)
    at C:\Users\steveo\AppData\Local\atom\app-1.2.4\resources\app.asar\src\view-registry.js:3:61
```

### Commands

```
     -2:44.8.0 autocomplete-plus:cancel (atom-text-editor.editor.is-focused.autocomplete-active)
  4x -2:44.3.0 editor:move-to-beginning-of-word (atom-text-editor.editor.is-focused)
     -2:43.6.0 editor:move-to-end-of-word (atom-text-editor.editor.is-focused)
     -2:42.7.0 core:move-right (atom-text-editor.editor.is-focused)
     -2:37.4.0 editor:move-to-end-of-word (atom-text-editor.editor.is-focused)
     -2:24.8.0 core:move-left (atom-text-editor.editor.is-focused)
     -2:13.5.0 core:copy (atom-text-editor.editor.is-focused)
     -1:37 core:save (atom-text-editor.editor.is-focused)
  2x -1:30.2.0 editor:newline (atom-text-editor.editor.is-focused)
     -1:22 autocomplete-plus:confirm (atom-text-editor.editor.is-focused.autocomplete-active)
     -1:21.7.0 editor:newline (atom-text-editor.editor.is-focused)
     -0:53.9.0 linter:lint (atom-text-editor.editor)
  5x -0:40.6.0 editor:move-to-beginning-of-word (atom-text-editor.editor.is-focused)
  6x -0:39.6.0 core:move-left (atom-text-editor.editor.is-focused)
  6x -0:26.3.0 editor:select-to-end-of-word (atom-text-editor.editor.is-focused)
     -0:16 core:backspace (atom-text-editor.editor.is-focused)
```

### Config

```json
{
  "core": {
    "themes": [
      "redmond-ui",
      "redmond-syntax"
    ]
  },
  "editor": {
    "invisibles": {}
  }
}
```

### Installed Packages

```coffee
# User
.bin, vundefined
atom-typescript, v7.12.0
linter, v1.11.2
redmond-syntax, v0.2.16
redmond-ui, v0.1.5

# Dev
No dev packages
```

**Comment by Ingramz** [comment]
Steps to reproduce:
1. Create horizontal scrollbar with bunch of newlines (in my case 35 lines)
2. Create a long enough line that horizontal scrollbar would appear
3. Delete/add characters to toggle horizontal scrollbar appearing/disappearing
4. Error message should appear

That 35 lines seems important and the behavior repeats every 6 lines for me (41, 47, 53, 59...)

**Comment by as-cii** [comment]
> That 35 lines seems important and the behavior repeats every 6 lines for me (41, 47, 53, 59...)

@Ingramz: thanks for this comment, it's really helpful! :pray: That's an interesting evidence, because `6` is the exact number of lines we put into each tile. Seems like when we're at that boundary, something goes wrong and causes a certain line number node not to be found. Unfortunately, I cannot reproduce the problem using your steps, but I wonder if there's any other variable in the equation that could trigger this behavior.

Could you point out which font are you using and if you have any specific styling configuration? Thanks!

**Comment by Ingramz** [comment]
@as-cii On my MacBook Pro running El Capitan with screen resolution 1280x800, I can reproduce this in fullscreen starting from line 41. No custom styling anywhere.

A sure way to trigger this would be just try it out around line 60-70 and then try the same thing on each line at a time. As far as I can figure it out, it has something to do with window height.

**Comment by Ingramz** [comment]
@as-cii it seems that sometimes the window height can be something that doesn't trigger the error. Changing it by a few pixels taller usually helps.

Also here's a screen capture.

![atombug](https://cloud.githubusercontent.com/assets/1017132/11807306/b5ba463e-a321-11e5-8f7a-f872be44dcc5.gif)

**Comment by as-cii** [comment]
That's helpful, @Ingramz! I am still unable to reproduce, though.

@izuzak: could you try reproducing this now that we have more information about it? :pray:

**Comment by Ingramz** [comment]
@as-cii between 1.4.0-beta3 and 1.5.0-beta0 something has been changed so that I am unable to reproduce this myself now. It's possible that it's fixed.

**Comment by as-cii** [comment]
> between 1.4.0-beta3 and 1.5.0-beta0 something has been changed so that I am unable to reproduce this myself now. It's possible that it's fixed.

Glad to hear that, @Ingramz. I'll leave this open for a while, just in case the error arises again. :+1:

**Comment by jasongilman** [comment]
I'm still seeing this error in Atom 1.4.1.

**Comment by lee-dohm** [comment]
@jasongilman Do you have specific repro steps or other information you can give?

**Comment by jasongilman** [comment]
Sorry it took me a while to respond and that I didn't have more information originally. I just saw the issue again. The stack trace is below. It hasn't happened for a week and I don't know exactly what causes it. I'm using my Atom package, Proto REPL, which quickly appends text to a text editor showing results from another running process. It may be that quickly adding text is the trigger here.

```
**Atom Version**: 1.4.2
**System**: Mac OS X 10.11.3


At /Users/jgilman/Applications/Atom.app/Contents/Resources/app.asar/src/line-numbers-tile-component.js:88

TypeError: Cannot read property 'hasOwnProperty' of undefined
    at LineNumbersTileComponent.module.exports.LineNumbersTileComponent.updateLineNumbers (/Users/jgilman/Applications/Atom.app/Contents/Resources/app.asar/src/line-numbers-tile-component.js:88:43)
    at LineNumbersTileComponent.module.exports.LineNumbersTileComponent.updateSync (/Users/jgilman/Applications/Atom.app/Contents/Resources/app.asar/src/line-numbers-tile-component.js:78:19)
    at LineNumberGutterComponent.module.exports.TiledComponent.updateTileNodes (/Users/jgilman/Applications/Atom.app/Contents/Resources/app.asar/src/tiled-component.js:67:19)
    at LineNumberGutterComponent.module.exports.TiledComponent.updateSync (/Users/jgilman/Applications/Atom.app/Contents/Resources/app.asar/src/tiled-component.js:30:12)
    at GutterContainerComponent.module.exports.GutterContainerComponent.updateSync (/Users/jgilman/Applications/Atom.app/Contents/Resources/app.asar/src/gutter-container-component.js:79:25)
    at TextEditorComponent.module.exports.TextEditorComponent.updateSync (/Users/jgilman/Applications/Atom.app/Contents/Resources/app.asar/src/text-editor-component.js:220:39)
    at /Users/jgilman/Applications/Atom.app/Contents/Resources/app.asar/src/text-editor-component.js:298:28
    at ViewRegistry.module.exports.ViewRegistry.performDocumentUpdate (/Users/jgilman/Applications/Atom.app/Contents/Resources/app.asar/src/view-registry.js:215:9)
    at /Users/jgilman/Applications/Atom.app/Contents/Resources/app.asar/src/view-registry.js:3:61
```

**Comment by Ingramz** [comment]
I can confirm that with 1.4.3 I can still reproduce what I've reported, but 1.5.0 betas don't have this issue. Please don't panic :smile_cat:

**Comment by jasongilman** [comment]
I missed that this was likely fixed in 1.5.0 beta. Thanks!
